docker/conf/log.config

```diff
@@ -12,7 +12,7 @@ handlers:
   file:
     class: logging.handlers.TimedRotatingFileHandler
     formatter: precise
-    filename: {{ LOG_FILE_PATH or "homeserver.log" }}
+    filename: {{ LOG_FILE_PATH or "/data/homeserver.log" }}
```

Comment:
the trouble with this is that it will still create an empty file in `/data`.

I think it would be better to surround the whole `file` and `buffer` configuration sections with `{% if LOG_FILE_PATH %}` (which means that the generated config will end up being similar to what we had before #9162).